extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -86,8 +104,17 @@ protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm) thro
     protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm, RunnableWithException afterEndTransaction)
             throws Exception {
 
+        TransactionTimeout timeoutAnnotation = getTransactionTimeout(ic);
+        if (timeoutAnnotation != null) {
+            tm.setTransactionTimeout(timeoutAnnotation.value());
+        }
         tm.begin();
         Transaction tx = tm.getTransaction();
+        if (timeoutAnnotation != null) {
+            //restore the default behaviour
+            tm.setTransactionTimeout(0);
```

Comment:
I think this should be done in a `finally` clause to be sure it's properly executed.

Reply:
Done